pyro/infer/trace_mmd.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import defaultdict
+
+import pyro
+import pyro.ops.jit
+from pyro import poutine
+from pyro.infer.trace_elbo import Trace_ELBO
+from pyro.infer.util import torch_item
+from pyro.util import warn_if_nan
+
+
+def _compute_mmd(X, Z, kernel):
+    mmd = kernel(X).mean() + kernel(Z).mean() - 2 * kernel(X, Z).mean()
+    return mmd
+
+
+class Trace_MMD(Trace_ELBO):
+    """
+    An objective similar to ELBO, but with Maximum Mean Discrepancy (MMD)
+    between marginal variational posterior and prior distributions
+    instead of KL-divergence between variational posterior and prior distributions
+    as in vanilla ELBO. See [1] for the corresponding variant of VAE model,
+    which is a special case of InfoVAE model [2].
+
+    :param kernel: A kernel used to compute MMD.
+        An instance of :class: `pyro.contrib.gp.kernels.kernel.Kernel`,
+        or a dict that maps latent variable names to instances of :class: `pyro.contrib.gp.kernels.kernel.Kernel`.
+        In the latter case, different kernels are used for different latent variables.
+
+    :param mmd_scale: A scaling factor for MMD terms.
+        Float, or a dict that maps latent variable names to floats.
+        In the latter case, different scaling factors are used for different latent variables.
+
+    References
+
+    [1] `A Tutorial on Information Maximizing Variational Autoencoders (InfoVAE)`
+        Shengjia Zhao
+        https://ermongroup.github.io/blog/a-tutorial-on-mmd-variational-autoencoders/
+
+    [2] `InfoVAE: Balancing Learning and Inference in Variational Autoencoders`
+        Shengjia Zhao, Jiaming Song, Stefano Ermon
+    """
+
+    def __init__(self,
+                 kernel, mmd_scale=1,
+                 num_particles=1,
+                 max_plate_nesting=float('inf'),
+                 max_iarange_nesting=None,  # DEPRECATED
+                 vectorize_particles=False,
+                 strict_enumeration_warning=True,
+                 ignore_jit_warnings=False,
+                 retain_graph=None):
+        super().__init__(
```

Comment:
Yep, python2 tests failed because of missed arguments in super. Why did python3 tests failed is a mystery for me, because everything is ok at my local machine. Trying to figure out the reason

Reply:
Seems that I have fixed it